lib/explorer/series.ex

```diff
@@ -3395,8 +3427,18 @@ defmodule Explorer.Series do
   defp cast_to_add({:datetime, p, tz}, {:duration, p}), do: {:datetime, p, tz}
   defp cast_to_add({:duration, p}, {:datetime, p, tz}), do: {:datetime, p, tz}
   defp cast_to_add({:duration, p}, {:duration, p}), do: {:duration, p}
+
+  defp cast_to_add({:decimal, p1, s1}, {:decimal, p2, s2}),
+    do: {:decimal, maybe_max(p1, p2), maybe_max(s1, s2)}
```

Comment:
Is there some rule we need to follow here? For example, is there a maximum value for precision and scale?

Reply:
According to ChatGPT:

---

Yes, in Apache Arrow's decimal128 type, the precision and scale are constrained as follows:

    Precision: This represents the total number of digits that can be stored, both before and after the decimal point. For decimal128, the maximum precision is 38 digits. This means that it can store up to 38 significant digits.

    Scale: This defines how many of the digits are allocated to the fractional part (i.e., after the decimal point). The scale can be any value between 0 and the precision value. For example, if you have a precision of 38 and set a scale of 10, then 28 digits can be used before the decimal point and 10 digits after.

Thus, the maximum precision is 38, and the scale can be anywhere from 0 to 38, depending on the application needs.

--- 

So I think we are good, but I'd encapsulate this logic in a function. :)